ui-speedspacechart/src/components/helpers/drawElements/speedLimitTags.ts

```diff
@@ -0,0 +1,183 @@
+import { Store, tooltipInfos } from '../../../types/chartTypes';
+import {
+  clearCanvas,
+  createSvgBlobUrl,
+  drawLinearLayerBackground,
+  drawRoundedRect,
+  drawSeparatorLinearLayer,
+  loadSvgImage,
+  maxPositionValues,
+  positionOnGraphScale,
+} from '../../utils';
+import {
+  LINEAR_LAYERS_HEIGHTS,
+  LINEAR_LAYERS_BACKGROUND_COLOR,
+  MARGINS,
+  colorDictionary,
+} from '../../const';
+import { drawSvgImageWithColor } from '../../utils';
+
+const questionSvg =
+  '<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path d="M0 8a8 8 0 1 1 16 0A8 8 0 0 1 0 8m8-6.5a6.5 6.5 0 1 0 0 13 6.5 6.5 0 0 0 0-13M6.92 6.085h.001a.75.75 0 1 1-1.342-.67c.169-.339.436-.701.849-.977C6.845 4.16 7.369 4 8 4a2.76 2.76 0 0 1 1.637.525c.503.377.863.965.863 1.725 0 .448-.115.83-.329 1.15-.205.307-.47.513-.692.662-.109.072-.22.138-.313.195l-.006.004a6 6 0 0 0-.26.16 1 1 0 0 0-.276.245.75.75 0 0 1-1.248-.832c.184-.264.42-.489.692-.661q.154-.1.313-.195l.007-.004c.1-.061.182-.11.258-.161a1 1 0 0 0 .277-.245C8.96 6.514 9 6.427 9 6.25a.61.61 0 0 0-.262-.525A1.27 1.27 0 0 0 8 5.5c-.369 0-.595.09-.74.187-.146.1-.263.238-.34.398M9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0"/></svg>';
+const alertFillSvg =
+  '<svg xmlns="http://www.w3.org/2000/svg" width="16" height="16" viewBox="0 0 16 16"><path d="M6.457 1.047c.659-1.234 2.427-1.234 3.086 0l6.082 11.378A1.75 1.75 0 0 1 14.082 15H1.918a1.75 1.75 0 0 1-1.543-2.575zm1.763.707a.25.25 0 0 0-.44 0L1.698 13.132a.25.25 0 0 0 .22.368h12.164a.25.25 0 0 0 .22-.368zm.53 3.996v2.5a.75.75 0 0 1-1.5 0v-2.5a.75.75 0 0 1 1.5 0M9 11a1 1 0 1 1-2 0 1 1 0 0 1 2 0"/></svg>';
+
+const questionBlobUrl = createSvgBlobUrl(questionSvg);
+const alertFillBlobUrl = createSvgBlobUrl(alertFillSvg);
+
+export const drawSpeedLimitTags = async (
+  ctx: CanvasRenderingContext2D,
+  width: number,
+  marginTop: number,
+  store: Store
+): Promise<tooltipInfos | null> => {
+  const {
+    speedLimitTags,
+    ratioX,
+    leftOffset,
+    layersDisplay: { electricalProfiles, powerRestrictions },
+    cursor,
+  } = store;
+
+  const { MARGIN_TOP, MARGIN_BOTTOM, MARGIN_LEFT, MARGIN_RIGHT } = MARGINS;
+
+  if (!speedLimitTags) return null;
```

Comment:
You don't need this because if speedLimitTags are undefined, you won't be able to activate the panel checkbox. See https://github.com/OpenRailAssociation/osrd-ui/blob/dev/ui-speedspacechart/src/components/common/SettingsPanel.tsx#L49

```suggestion
```

Reply:
good catch